pkg/binding/write.go

```diff
@@ -86,13 +86,13 @@ func Write(
 		}
 	}
 
-	var e event.Event
+	var e *event.Event
 	e, err = ToEvent(ctx, message, transformers)
 	if err != nil {
 		return enc, err
 	}
 
-	message = EventMessage(e)
+	message = EventMessage(*e)
```

Comment:
This provokes a copy, you should do `(*EventMessage)(e)` to just cast it as a pointer